### PR TITLE
Use a case-insensitive search for UCR filtering

### PR DESCRIPTION
**Pull request by esoergel**
(paired with @orangejenny)
http://manage.dimagi.com/default.asp?188776
@czue @sravfeyn

### CODE REVIEW

**Comment by czue** [comment]
:+1: seems legit

**Comment by czue** [comment]
haven't reviewed https://github.com/dimagi/commcare-hq/pull/9423 too closely yet but depending how hard it is could be nice to add a test case for this

File: corehq/apps/userreports/reports/filters/choice_providers.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def _sql_column(self):
     def get_values_for_query(self, query_context):
         query = self._adapter.session_helper.Session.query(self._sql_column)
         if query_context.query:
-            query = query.filter(self._sql_column.contains(query_context.query))
+            query = query.filter(self._sql_column.ilike(u"%{}%".format(query_context.query)))
```
Comment (sravfeyn) [inline corehq/apps/userreports/reports/filters/choice_providers.py]: (No blocker) Using `icontains` will give same result without having to use match strings.

Comment (czue) [inline corehq/apps/userreports/reports/filters/choice_providers.py]: :+1: nice, good point

**Comment by NoahCarnahan** [comment]
@orangejenny @esoergel 
How confident are you that `ilike()` will escape the query against sql injections?
I had actually looked into this issue a few weeks ago but held off on doing anything because
it seems like `ilike()` might not do any escaping to its argument. I have that impression
because the pattern matching string would have special characters in it, where as `contains()`
is expecting text. @nickpell might have insight on this question too.

File: corehq/apps/userreports/reports/filters/choice_providers.py (hunk 1 above)
Comment (esoergel) [inline corehq/apps/userreports/reports/filters/choice_providers.py]: That's what we were hoping for - unfortunately SQLAlchemy doesn't provide that method:
```python
ipdb> self._sql_column.icontains
*** AttributeError: Neither 'Column' object nor 'Comparator' object has an attribute 'icontains'
```
Here's the open request to add that: https://bitbucket.org/zzzeek/sqlalchemy/issues/3482/icontains-istartswith-iendswith

**Comment by orangejenny** [comment]
@NoahCarnahan honestly, I would expect that part of the point of SQLAlchemy would be that it'd escape against SQL injection. It's only % that's a special character for like expressions, so it isn't a hard problem.

This suggests SQLAlchemy does handle escaping properly like a decent ORM should ;)
https://rhaptos2user.readthedocs.org/en/latest/sqlinjection.html

**Comment by esoergel** [comment]
@NoahCarnahan Good thinking - I'm not sure.  My understanding is that `like("%{}%".format(other))` is literally the same as `contains(other)`, and `ilike` just adds case-insensitivity, so I'd expect them to behave however that does in that regard.

**Comment by NoahCarnahan** [comment]
Cool, I'm convinced  :)

File: corehq/apps/userreports/reports/filters/choice_providers.py (hunk 1 above)
Comment (sravfeyn) [inline corehq/apps/userreports/reports/filters/choice_providers.py]: Ha, thanks for clarifying!

**Comment by nickpell** [comment]
@esoergel adding ```print query``` shows that a bind param is used, so we're safe from injection

**Comment by esoergel** [comment]
:fireworks: thanks bubbles!

**Pull request merged**